electron/BrowserRequestController.ts

```diff
@@ -0,0 +1,81 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+import { BrowserWindow } from 'electron';
+import { ipcMain as ipc } from 'electron-better-ipc';
+import logger from 'electron-log';
+import type { ClientBrowserRequest, RecordJourneyRequest } from '../common/types';
+import { isTestJourneyRequest } from '../src/common/shared';
+
+const isRecordJourneyRequest = (req: any): req is RecordJourneyRequest => !!req?.data?.url;
+
+export class BrowserRequestController {
```

Comment:
It seems like what this class does is just to keep track of whether the browser is running or not. Could this whole class be replaced by a boolean variable in execution?
```js
// electron/execution.ts
let isBrowserRunning= false;

async function onRecordJourneys(...args) {
  if (isBrowserRunning) {
   throw new Error('Cannot start recording a journey, a browser operation is already in progress.');
  } 
  isBrowserOpen = true;
   ...
}

async function onTest(...args) {
  if (isBrowserRunning) {
   throw new Error('Cannot start testing a journey, a browser operation is already in progress.');
  } 
  isBrowserOpen = true;
   ...
}
```

But please let me know if there are some important points that I missed 👍

Reply:
++ felt like we are over complicating things here by using controller class.

Reply:
I may have misinterpreted the original intention behind the issue. Certainly we can achieve this with a simple boolean instead, sorry for over-engineering this.

Reply:
Sorry for the misleading issue - I should've revisited it and updated it as some of the assumptions/suggestions I had back then were not that intuitive and well-thought through.

Reply:
No worries. I have pushed the commits with the class/tests to my remote so if we ever want to enhance this limiting to allow simultaneous test runs/recording sessions we can draw on that code for inspiration.